### PR TITLE
Rework backend to be more efficient

### LINKED ISSUE #85: Create general competition servlet

### LINKED ISSUE #86: Write servlet for running ranking and networth db updates on cronjob

### LINKED ISSUE #87: Write servlet for returning user investments

### LINKED ISSUE #88: Write servlet for returning competitor info

### LINKED ISSUE #89: Write servlet to return recent buys

### PR DESCRIPTION
**Pull request by roonil-wazlib**
- Calculations for user rank and networth all now run by a ScheduledServlet, which is pinged by the Python script
 - Servlets reworked to only deliver required information and not perform ranking calculations

Servlets added:
 - RecentBuys servlet returns a list of recent purchases from other users in the same competition
 - CompetitionsServlet returns a list of competition summaries for all the competitions the user is in
 - SingleCompetitionServlet returns this information for just one competition, eg in case of a refresh of competition page
 - CompetitorsServlet returns a list of relevant information about all the other competitors in this competition

Objects added:
 - BasicCompetitor is a minimalist competitor object used only for ranking purposes in the ScheduledServlet
 - CompetitionSummary contains key info about a competition, for purposes of the scrolling competition dashboard display
 - CompetitorInfo contains networth and investment information about a competitor
 - RecentBuy contains summary information about a purchase

Resolves #85, resolves #86, resolves #87, resolves #88, resolves #89

### CODE REVIEW

File: project/src/main/java/com/google/sps/data/CompetitionSummary.java
Comment (lin8231) [inline project/src/main/java/com/google/sps/data/CompetitionSummary.java:24]: Wait I thought rankYesterday wasn't possible - is this still a thing?

Comment (lin8231) [reply project/src/main/java/com/google/sps/data/CompetitionSummary.java:24]: Or has the rework made this possible

File: project/src/main/java/com/google/sps/servlets/CompetitorsServlet.java
Comment (lin8231) [inline project/src/main/java/com/google/sps/servlets/CompetitorsServlet.java:40]: This comment looks to be outdated

**Review by lin8231 — COMMENTED** [review]
Just to clarify, is the investments servlet supposed to serve 
* List of User’s UNSOLD sGonks. Each has:
    * Search term
    * Date invested
    * Amount invested
    * Price now 
    * Price yesterday
OR
* History of price of sGonks owned (maybe include sold ones too, if it’s possible to display chart lines that cut off midway?) I would imagine this is also a list, each list item is:
    * sGonk name
    * A list of this sGonk’s price over time (since the person bought it)
OR both? 


Also - I'm assuming this: 
* History of all competitor ranks (for the chart): I imagine this is a list, where each list item is:
    * Person name
    * A list of this person’s net worth over time 
Doesn't yet exist, right?

**Comment by lin8231** [comment]
Screen Shot 2021-01-29 at 5 34 15 pm
The lack of colours makes this difficult but here for clarity - the crossed out ones are LGTM, the bolded ones just need clarification, the ones with no text decorations I assume don't exist at this point in time. Is this correct?

**Comment by lin8231** [comment]
Also can u pls fix the merge conflicts before merge

File: project/src/main/java/com/google/sps/data/CompetitionSummary.java
Comment (roonil-wazlib) [reply project/src/main/java/com/google/sps/data/CompetitionSummary.java:24]: Yeah I can do it now after the adjustments I made

**Comment by roonil-wazlib** [comment]
> Screen Shot 2021-01-29 at 5 34 15 pm
> The lack of colours makes this difficult but here for clarity - the crossed out ones are LGTM, the bolded ones just need clarification, the ones with no text decorations I assume don't exist at this point in time. Is this correct?

I have a servlet named InvestmentsServlet already in master. It returns info about every investment owned by the current user, including amt invested, investment date, sell date, current worth, as well as a list of all the data points on it starting from a week prior to investment date. In the case where the investment has not been sold, sell date is returned as 0. I'm going to assume this is sufficient for both the history of price of sgonks owned bullet point, as well as the list of unsold sgonks - all you have to do for that one is filter by the sell date.

**Comment by roonil-wazlib** [comment]
> I'm assuming this:
> 
> * History of all competitor ranks (for the chart): I imagine this is a list, where each list item is:
>   
>   * Person name
>   * A list of this person’s net worth over time
>     Doesn't yet exist, right?

no it doesn't. It would require me to maintain time series trends for user networth in datastore along with the trends data. It's doable but wasn't in the project plan so I didn't code it up. From my point of view it's not an essential feature, WDYT?

**Comment by lin8231** [comment]
> > Screen Shot 2021-01-29 at 5 34 15 pm
> > The lack of colours makes this difficult but here for clarity - the crossed out ones are LGTM, the bolded ones just need clarification, the ones with no text decorations I assume don't exist at this point in time. Is this correct?
> 
> I have a servlet named InvestmentsServlet already in master. It returns info about every investment owned by the current user, including amt invested, investment date, sell date, current worth, as well as a list of all the data points on it starting from a week prior to investment date. In the case where the investment has not been sold, sell date is returned as 0. I'm going to assume this is sufficient for both the history of price of sgonks owned bullet point, as well as the list of unsold sgonks - all you have to do for that one is filter by the sell date.

Ah okay - so filtering for unsold sGonks for the "currently owned investments" will be done on my end, I see. That can be done easily enough. 
Will this be a problem with querying for the "currently owned investments" when a user is deep into a competition and has waaaay more sold sgonks than unsold sgonks?

**Comment by lin8231** [comment]
> > > Screen Shot 2021-01-29 at 5 34 15 pm
> > > The lack of colours makes this difficult but here for clarity - the crossed out ones are LGTM, the bolded ones just need clarification, the ones with no text decorations I assume don't exist at this point in time. Is this correct?
> > 
> > 
> > I have a servlet named InvestmentsServlet already in master. It returns info about every investment owned by the current user, including amt invested, investment date, sell date, current worth, as well as a list of all the data points on it starting from a week prior to investment date. In the case where the investment has not been sold, sell date is returned as 0. I'm going to assume this is sufficient for both the history of price of sgonks owned bullet point, as well as the list of unsold sgonks - all you have to do for that one is filter by the sell date.
> 
> Ah okay - so filtering for unsold sGonks for the "currently owned investments" will be done on my end, I see. That can be done easily enough.
> Will this be a problem with querying for the "currently owned investments" when a user is deep into a competition and has waaaay more sold sgonks than unsold sgonks?

Oh and also, how do I get the price yesterday?

**Comment by lin8231** [comment]
> > I'm assuming this:
> > 
> > * History of all competitor ranks (for the chart): I imagine this is a list, where each list item is:
> >   
> >   * Person name
> >   * A list of this person’s net worth over time
> >     Doesn't yet exist, right?
> 
> no it doesn't. It would require me to maintain time series trends for user networth in datastore along with the trends data. It's doable but wasn't in the project plan so I didn't code it up. From my point of view it's not an essential feature, WDYT?

That would be for displaying this graph on the comp page:
Screen Shot 2021-01-29 at 5 52 05 pm

What do you suggest might be the best thing to display in that graph?

**Comment by roonil-wazlib** [comment]
> > > I'm assuming this:
> > > 
> > > * History of all competitor ranks (for the chart): I imagine this is a list, where each list item is:
> > >   
> > >   * Person name
> > >   * A list of this person’s net worth over time
> > >     Doesn't yet exist, right?
> > 
> > 
> > no it doesn't. It would require me to maintain time series trends for user networth in datastore along with the trends data. It's doable but wasn't in the project plan so I didn't code it up. From my point of view it's not an essential feature, WDYT?
> 
> That would be for displaying this graph on the comp page:
> Screen Shot 2021-01-29 at 5 52 05 pm
> 
> What do you suggest might be the best thing to display in that graph?

You could just graph today vs yesterday for now?

**Comment by lin8231** [comment]
> > > > I'm assuming this:
> > > > 
> > > > * History of all competitor ranks (for the chart): I imagine this is a list, where each list item is:
> > > >   
> > > >   * Person name
> > > >   * A list of this person’s net worth over time
> > > >     Doesn't yet exist, right?
> > > 
> > > 
> > > no it doesn't. It would require me to maintain time series trends for user networth in datastore along with the trends data. It's doable but wasn't in the project plan so I didn't code it up. From my point of view it's not an essential feature, WDYT?
> > 
> > 
> > That would be for displaying this graph on the comp page:
> > Screen Shot 2021-01-29 at 5 52 05 pm
> > What do you suggest might be the best thing to display in that graph?
> 
> You could just graph today vs yesterday for now?

Can do that for now yeah. Would look kinda sad but let's leave it to be something to improve on later if there is time.

**Comment by roonil-wazlib** [comment]
> > > > Screen Shot 2021-01-29 at 5 34 15 pm
> > > > The lack of colours makes this difficult but here for clarity - the crossed out ones are LGTM, the bolded ones just need clarification, the ones with no text decorations I assume don't exist at this point in time. Is this correct?
> > > 
> > > 
> > > I have a servlet named InvestmentsServlet already in master. It returns info about every investment owned by the current user, including amt invested, investment date, sell date, current worth, as well as a list of all the data points on it starting from a week prior to investment date. In the case where the investment has not been sold, sell date is returned as 0. I'm going to assume this is sufficient for both the history of price of sgonks owned bullet point, as well as the list of unsold sgonks - all you have to do for that one is filter by the sell date.
> > 
> > 
> > Ah okay - so filtering for unsold sGonks for the "currently owned investments" will be done on my end, I see. That can be done easily enough.
> > Will this be a problem with querying for the "currently owned investments" when a user is deep into a competition and has waaaay more sold sgonks than unsold sgonks?
> 
> Oh and also, how do I get the price yesterday?

I can't imagine they would accumulate massive amounts of investments from a computer's pov. We originally estimated on average 1 a day would be quite a few over a competition duration.

The price yesterday is just the second to last point in the list of data points.

**Comment by roonil-wazlib** [comment]
> > > > > I'm assuming this:
> > > > > 
> > > > > * History of all competitor ranks (for the chart): I imagine this is a list, where each list item is:
> > > > >   
> > > > >   * Person name
> > > > >   * A list of this person’s net worth over time
> > > > >     Doesn't yet exist, right?
> > > > 
> > > > 
> > > > no it doesn't. It would require me to maintain time series trends for user networth in datastore along with the trends data. It's doable but wasn't in the project plan so I didn't code it up. From my point of view it's not an essential feature, WDYT?
> > > 
> > > 
> > > That would be for displaying this graph on the comp page:
> > > Screen Shot 2021-01-29 at 5 52 05 pm
> > > What do you suggest might be the best thing to display in that graph?
> > 
> > 
> > You could just graph today vs yesterday for now?
> 
> Can do that for now yeah. Would look kinda sad but let's leave it to be something to improve on later if there is time.

Yeah I can add it easily enough, but it's not important for functionality so I'd rather not spend time on it right now.

**Comment by lin8231** [comment]
> > > > > Screen Shot 2021-01-29 at 5 34 15 pm
> > > > > The lack of colours makes this difficult but here for clarity - the crossed out ones are LGTM, the bolded ones just need clarification, the ones with no text decorations I assume don't exist at this point in time. Is this correct?
> > > > 
> > > > 
> > > > I have a servlet named InvestmentsServlet already in master. It returns info about every investment owned by the current user, including amt invested, investment date, sell date, current worth, as well as a list of all the data points on it starting from a week prior to investment date. In the case where the investment has not been sold, sell date is returned as 0. I'm going to assume this is sufficient for both the history of price of sgonks owned bullet point, as well as the list of unsold sgonks - all you have to do for that one is filter by the sell date.
> > > 
> > > 
> > > Ah okay - so filtering for unsold sGonks for the "currently owned investments" will be done on my end, I see. That can be done easily enough.
> > > Will this be a problem with querying for the "currently owned investments" when a user is deep into a competition and has waaaay more sold sgonks than unsold sgonks?
> > 
> > 
> > Oh and also, how do I get the price yesterday?
> 
> I can't imagine they would accumulate massive amounts of investments from a computer's pov. We originally estimated on average 1 a day would be quite a few over a competition duration.
> 
> The price yesterday is just the second to last point in the list of data points.

SGTM to the unsold sgonks thing. 

As for price yesterday - I think I read in the code that the price (today/current price) has to be calculated using the today datapoint - since I need the price yesterday (as opposed to raw popularity) will I need to calculate that as well? 
Or, are the datapoints all prices that have already been calculated?

**Comment by roonil-wazlib** [comment]
> > > > > > Screen Shot 2021-01-29 at 5 34 15 pm
> > > > > > The lack of colours makes this difficult but here for clarity - the crossed out ones are LGTM, the bolded ones just need clarification, the ones with no text decorations I assume don't exist at this point in time. Is this correct?
> > > > > 
> > > > > 
> > > > > I have a servlet named InvestmentsServlet already in master. It returns info about every investment owned by the current user, including amt invested, investment date, sell date, current worth, as well as a list of all the data points on it starting from a week prior to investment date. In the case where the investment has not been sold, sell date is returned as 0. I'm going to assume this is sufficient for both the history of price of sgonks owned bullet point, as well as the list of unsold sgonks - all you have to do for that one is filter by the sell date.
> > > > 
> > > > 
> > > > Ah okay - so filtering for unsold sGonks for the "currently owned investments" will be done on my end, I see. That can be done easily enough.
> > > > Will this be a problem with querying for the "currently owned investments" when a user is deep into a competition and has waaaay more sold sgonks than unsold sgonks?
> > > 
> > > 
> > > Oh and also, how do I get the price yesterday?
> > 
> > 
> > I can't imagine they would accumulate massive amounts of investments from a computer's pov. We originally estimated on average 1 a day would be quite a few over a competition duration.
> > The price yesterday is just the second to last point in the list of data points.
> 
> SGTM to the unsold sgonks thing.
> 
> As for price yesterday - I think I read in the code that the price (today/current price) has to be calculated using the today datapoint - since I need the price yesterday (as opposed to raw popularity) will I need to calculate that as well?
> Or, are the datapoints all prices that have already been calculated?

The datapoints list the popularity over time, out of 100. You are correct that they can change as new days of data accumulate, but what doesn't change is their position relative to eachother. So, for example, if a user buys in on a day that has popularity 50 and invests 100, and then on a day in the future popularity is 100, they can cash out with 200. If they don't cash out and instead wait another day that ends up being more popular than the previous day, the datapoints all adjust proportionately. So the new day would be 100, the previous day might look more like 80, and the the day they bought in would look like 40 instead of 50. That's still double what they cashed in on either way.

I should confirm what you mean by price yesterday though. The investments themselves don't have a 'unit price' like a stock does, so the only data you have is the popularity score which is always going to be relative, or the amount they could have cashed out on that day, which is constant but would require you to do some basic percentage calculations. It depends what you're planning on using this data point for I guess, I'm not too sure what the issue is.

**Comment by lin8231** [comment]
> > > > > > > Screen Shot 2021-01-29 at 5 34 15 pm
> > > > > > > The lack of colours makes this difficult but here for clarity - the crossed out ones are LGTM, the bolded ones just need clarification, the ones with no text decorations I assume don't exist at this point in time. Is this correct?
> > > > > > 
> > > > > > 
> > > > > > I have a servlet named InvestmentsServlet already in master. It returns info about every investment owned by the current user, including amt invested, investment date, sell date, current worth, as well as a list of all the data points on it starting from a week prior to investment date. In the case where the investment has not been sold, sell date is returned as 0. I'm going to assume this is sufficient for both the history of price of sgonks owned bullet point, as well as the list of unsold sgonks - all you have to do for that one is filter by the sell date.
> > > > > 
> > > > > 
> > > > > Ah okay - so filtering for unsold sGonks for the "currently owned investments" will be done on my end, I see. That can be done easily enough.
> > > > > Will this be a problem with querying for the "currently owned investments" when a user is deep into a competition and has waaaay more sold sgonks than unsold sgonks?
> > > > 
> > > > 
> > > > Oh and also, how do I get the price yesterday?
> > > 
> > > 
> > > I can't imagine they would accumulate massive amounts of investments from a computer's pov. We originally estimated on average 1 a day would be quite a few over a competition duration.
> > > The price yesterday is just the second to last point in the list of data points.
> > 
> > 
> > SGTM to the unsold sgonks thing.
> > As for price yesterday - I think I read in the code that the price (today/current price) has to be calculated using the today datapoint - since I need the price yesterday (as opposed to raw popularity) will I need to calculate that as well?
> > Or, are the datapoints all prices that have already been calculated?
> 
> The datapoints list the popularity over time, out of 100. You are correct that they can change as new days of data accumulate, but what doesn't change is their position relative to eachother. So, for example, if a user buys in on a day that has popularity 50 and invests 100, and then on a day in the future popularity is 100, they can cash out with 200. If they don't cash out and instead wait another day that ends up being more popular than the previous day, the datapoints all adjust proportionately. So the new day would be 100, the previous day might look more like 80, and the the day they bought in would look like 40 instead of 50. That's still double what they cashed in on either way.
> 
> I should confirm what you mean by price yesterday though. The investments themselves don't have a 'unit price' like a stock does, so the only data you have is the popularity score which is always going to be relative, or the amount they could have cashed out on that day, which is constant but would require you to do some basic percentage calculations. It depends what you're planning on using this data point for I guess, I'm not too sure what the issue is.

Thanks for the clarification. The intention of getting priceYesterday was so that there can be a +t$xxx or a -t$xxx value next to some displays of the currentPrice to compare what happened to your sGonk over night. But from what it sounds like I can calculate the price yesterday by doing a ratio of yesterday and buy-in date's popularity scores and multiplying it by the amount invested, right?

**Comment by roonil-wazlib** [comment]
> > > > > > > > Screen Shot 2021-01-29 at 5 34 15 pm
> > > > > > > > The lack of colours makes this difficult but here for clarity - the crossed out ones are LGTM, the bolded ones just need clarification, the ones with no text decorations I assume don't exist at this point in time. Is this correct?
> > > > > > > 
> > > > > > > 
> > > > > > > I have a servlet named InvestmentsServlet already in master. It returns info about every investment owned by the current user, including amt invested, investment date, sell date, current worth, as well as a list of all the data points on it starting from a week prior to investment date. In the case where the investment has not been sold, sell date is returned as 0. I'm going to assume this is sufficient for both the history of price of sgonks owned bullet point, as well as the list of unsold sgonks - all you have to do for that one is filter by the sell date.
> > > > > > 
> > > > > > 
> > > > > > Ah okay - so filtering for unsold sGonks for the "currently owned investments" will be done on my end, I see. That can be done easily enough.
> > > > > > Will this be a problem with querying for the "currently owned investments" when a user is deep into a competition and has waaaay more sold sgonks than unsold sgonks?
> > > > > 
> > > > > 
> > > > > Oh and also, how do I get the price yesterday?
> > > > 
> > > > 
> > > > I can't imagine they would accumulate massive amounts of investments from a computer's pov. We originally estimated on average 1 a day would be quite a few over a competition duration.
> > > > The price yesterday is just the second to last point in the list of data points.
> > > 
> > > 
> > > SGTM to the unsold sgonks thing.
> > > As for price yesterday - I think I read in the code that the price (today/current price) has to be calculated using the today datapoint - since I need the price yesterday (as opposed to raw popularity) will I need to calculate that as well?
> > > Or, are the datapoints all prices that have already been calculated?
> > 
> > 
> > The datapoints list the popularity over time, out of 100. You are correct that they can change as new days of data accumulate, but what doesn't change is their position relative to eachother. So, for example, if a user buys in on a day that has popularity 50 and invests 100, and then on a day in the future popularity is 100, they can cash out with 200. If they don't cash out and instead wait another day that ends up being more popular than the previous day, the datapoints all adjust proportionately. So the new day would be 100, the previous day might look more like 80, and the the day they bought in would look like 40 instead of 50. That's still double what they cashed in on either way.
> > I should confirm what you mean by price yesterday though. The investments themselves don't have a 'unit price' like a stock does, so the only data you have is the popularity score which is always going to be relative, or the amount they could have cashed out on that day, which is constant but would require you to do some basic percentage calculations. It depends what you're planning on using this data point for I guess, I'm not too sure what the issue is.
> 
> Thanks for the clarification. The intention of getting priceYesterday was so that there can be a +t$xxx or a -t$xxx value next to some displays of the currentPrice to compare what happened to your sGonk over night. But from what it sounds like I can calculate the price yesterday by doing a ratio of yesterday and buy-in date's popularity scores and multiplying it by the amount invested, right? (and I assume the buy-in date's popularity score is simply the index 0 on the list?

Easiest to do it relative to today's data point. priceYesterday = currentValue * (popularityYesterday / popularityToday). Don't even need to backtrack to investment date.

**Comment by lin8231** [comment]
> > > > > > > > > Screen Shot 2021-01-29 at 5 34 15 pm
> > > > > > > > > The lack of colours makes this difficult but here for clarity - the crossed out ones are LGTM, the bolded ones just need clarification, the ones with no text decorations I assume don't exist at this point in time. Is this correct?
> > > > > > > > 
> > > > > > > > 
> > > > > > > > I have a servlet named InvestmentsServlet already in master. It returns info about every investment owned by the current user, including amt invested, investment date, sell date, current worth, as well as a list of all the data points on it starting from a week prior to investment date. In the case where the investment has not been sold, sell date is returned as 0. I'm going to assume this is sufficient for both the history of price of sgonks owned bullet point, as well as the list of unsold sgonks - all you have to do for that one is filter by the sell date.
> > > > > > > 
> > > > > > > 
> > > > > > > Ah okay - so filtering for unsold sGonks for the "currently owned investments" will be done on my end, I see. That can be done easily enough.
> > > > > > > Will this be a problem with querying for the "currently owned investments" when a user is deep into a competition and has waaaay more sold sgonks than unsold sgonks?
> > > > > > 
> > > > > > 
> > > > > > Oh and also, how do I get the price yesterday?
> > > > > 
> > > > > 
> > > > > I can't imagine they would accumulate massive amounts of investments from a computer's pov. We originally estimated on average 1 a day would be quite a few over a competition duration.
> > > > > The price yesterday is just the second to last point in the list of data points.
> > > > 
> > > > 
> > > > SGTM to the unsold sgonks thing.
> > > > As for price yesterday - I think I read in the code that the price (today/current price) has to be calculated using the today datapoint - since I need the price yesterday (as opposed to raw popularity) will I need to calculate that as well?
> > > > Or, are the datapoints all prices that have already been calculated?
> > > 
> > > 
> > > The datapoints list the popularity over time, out of 100. You are correct that they can change as new days of data accumulate, but what doesn't change is their position relative to eachother. So, for example, if a user buys in on a day that has popularity 50 and invests 100, and then on a day in the future popularity is 100, they can cash out with 200. If they don't cash out and instead wait another day that ends up being more popular than the previous day, the datapoints all adjust proportionately. So the new day would be 100, the previous day might look more like 80, and the the day they bought in would look like 40 instead of 50. That's still double what they cashed in on either way.
> > > I should confirm what you mean by price yesterday though. The investments themselves don't have a 'unit price' like a stock does, so the only data you have is the popularity score which is always going to be relative, or the amount they could have cashed out on that day, which is constant but would require you to do some basic percentage calculations. It depends what you're planning on using this data point for I guess, I'm not too sure what the issue is.
> > 
> > 
> > Thanks for the clarification. The intention of getting priceYesterday was so that there can be a +t$xxx or a -t$xxx value next to some displays of the currentPrice to compare what happened to your sGonk over night. But from what it sounds like I can calculate the price yesterday by doing a ratio of yesterday and buy-in date's popularity scores and multiplying it by the amount invested, right? (and I assume the buy-in date's popularity score is simply the index 0 on the list?
> 
> Easiest to do it relative to today's data point. priceYesterday = currentValue * (popularityYesterday / popularityToday). Don't even need to backtrack to investment date.
> 
> Technically buy in date is index 7, I'm providing a week of context data for new sgonks otherwise there will be nothing there. But yeah, you don't need that datapoint anyway.

Yup SGTM.

**Review by lin8231 — APPROVED** [review]

File: project/src/main/java/com/google/sps/servlets/CompetitorsServlet.java
Comment (leonnemets) [inline project/src/main/java/com/google/sps/servlets/CompetitorsServlet.java:110]: can this be an SQL? you are retrieving lots ie (*) from DB.

something like select count(*) FROM investmen.....

you should get back the count.

File: project/src/main/java/com/google/sps/servlets/RecentBuysServlet.java
Comment (leonnemets) [inline project/src/main/java/com/google/sps/servlets/RecentBuysServlet.java:71]: ORDER BY is nice.

Instead of limiting in Java you can limit in SQL, let me know if you need a hand finding how

File: project/src/main/java/com/google/sps/servlets/ScheduledServlet.java
Comment (leonnemets) [inline project/src/main/java/com/google/sps/servlets/ScheduledServlet.java:86]: {nit} format

**Review by leonnemets — APPROVED** [review]
Nice work, just address the comments

File: project/src/main/java/com/google/sps/servlets/CompetitionsServlet.java
Comment (roonil-wazlib) [reply project/src/main/java/com/google/sps/servlets/CompetitionsServlet.java:52]: This is a good point. I will create an issue for it an address in a separate PR instead of adding more modified files to this one.

**Pull request merged**
